### PR TITLE
Implement blocks endpoint

### PR DESCRIPTION
**Pull request by diminator**
implements `GET /blocks/{block_id}` and `GET /blocks?tx_id={tx_id}` according to #830

(only review after #1026 is implemented)

- [x] `GET /blocks/{block_id}`
- [x] `GET /blocks?tx_id={tx_id}`
- [x] tests
- [x] update docs in #830

### CODE REVIEW

File: bigchaindb/web/views/blocks.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            block_id (str): the id of the block.
+
+        Return:
+            A JSON string containing the data about the block.
+        """
+
+        pool = current_app.config['bigchain_pool']
+
+        with pool() as bigchain:
+            block = bigchain.get_block(block_id=block_id)
```
Comment (TimDaub) [inline bigchaindb/web/views/blocks.py]: According to [the specification](https://github.com/bigchaindb/bigchaindb/pull/830/files#diff-9cd1814988545a4c7fb7457a3f928312R308) this endpoint should take a query parameter `?status`.

For more detail see spec.

Comment (diminator) [inline bigchaindb/web/views/blocks.py]: still updating the spec :-)

Comment (diminator) [inline bigchaindb/web/views/blocks.py]: resolved

File: bigchaindb/web/views/blocks.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        status = args['status']
+
+        pool = current_app.config['bigchain_pool']
+
+        with pool() as bigchain:
+            blocks = bigchain.get_blocks_status_containing_tx(tx_id)
+            if blocks:
+                if status:
+                    blocks = {k: v for k, v in blocks.items() if v == status}
+                blocks = list(blocks.keys())
```
Comment (sohkai) [inline bigchaindb/web/views/blocks.py]: You *could* reformulate this as a list comprehension, to avoid the nested `if`s (imo, makes it more complex):

```python
block_statuses = bigchain.get_blocks_status_containing_tx(tx_id)
blocks = [block_id for block_id, block_status in block_statuses.items() if not status or block_status == status]
```

File: tests/web/test_blocks.py [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+        'message': 'Unknown arguments: foo'
+    }
+
+    res = client.get(BLOCKS_ENDPOINT + "?tx_id=123&status=123")
+    assert res.status_code == 400
+    assert res.json == {
+        'message': {
+            'status': '123 is not a valid choice'
+        }
+    }
```
Comment (sohkai) [inline tests/web/test_blocks.py]: Another good test here would be an empty `tx_id`; I tried it and it doesn't complain (just returns an empty list) -- I assume the reqparser just returns an empty string for that argument.

E.g.:

```python
client.get(BLOCKS_ENDPOINT + "?tx_id=")
```
